jib-core/src/test/java/com/google/cloud/tools/jib/cache/CacheReaderTest.java

```diff
@@ -134,28 +133,29 @@ public void testGetUpToDateLayerBySourceFiles()
       Assert.assertEquals(3, cachedLayers.size());
       classesCachedLayer = cachedLayers.get(2);
 
+      Assert.assertNotNull(classesCachedLayer.getMetadata());
       classesCachedLayer
           .getMetadata()
-          .setEntries(
-              ImmutableList.of(
-                  new LayerMetadata.LayerMetadataEntry(
-                      Collections.singletonList(testSourceFiles.toString()), "extractionPath")));
+          .setEntry(ImmutableList.of(testSourceFiles.toString()), "/some/extraction/path");
     }
 
     try (Cache cache = Cache.init(testCacheFolder)) {
       CacheReader cacheReader = new CacheReader(cache);
 
+      ImmutableList<LayerEntry> layerEntryFilter =
+          ImmutableList.of(
+              new LayerEntry(ImmutableList.of(testSourceFiles), "/some/extraction/path"));
+
+      CachedLayerWithMetadata upToDateLayer =
+          cacheReader.getUpToDateLayerByLayerEntries(layerEntryFilter);
```

Comment:
Aside: this `layerEntryFilter` doesnt seem like a filter (esp given that there is a LayerFilter class), and `getUpToDateLayerByLayerEntries()` seems more `WithLayerEntries()`?  The filter made me think that this could return a cached layer that contained *more* than these layer entries.

Reply:
Ah yes, that's a bad naming. Will change.